From 0e56aeae8c5d832e877592bbc57643161dce74b4 Mon Sep 17 00:00:00 2001
From: Sravan Voleti <sravankumar@codeaurora.org>
Date: Thu, 30 Aug 2018 11:03:25 +0530
Subject: [PATCH 34/38] BLE: Fix NPE during start advertising

Use case:
Open any app that performs advertising.

Expected result:
BLE should Start advertisement.

Observed result :
Observed force close in BA App.

Fix:
set zero into parcel if transport
discovery data is empty.

Test: BLE Start advertisement successes after performing above use case.

CRs-Fixed: 2304532
Change-Id: Ica730f8d22367864308412621a8d87e625198bec
---
 core/java/android/bluetooth/le/AdvertiseData.java | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/java/android/bluetooth/le/AdvertiseData.java b/core/java/android/bluetooth/le/AdvertiseData.java
index 21bc0d35..d7f22bf4 100644
--- a/core/java/android/bluetooth/le/AdvertiseData.java
+++ b/core/java/android/bluetooth/le/AdvertiseData.java
@@ -170,8 +170,8 @@ public final class AdvertiseData implements Parcelable {
         }
         dest.writeByte((byte) (getIncludeTxPowerLevel() ? 1 : 0));
         dest.writeByte((byte) (getIncludeDeviceName() ? 1 : 0));
-        if(mTransportDiscoveryData != null) {
-            dest.writeInt(mTransportDiscoveryData.length);
+        dest.writeInt(mTransportDiscoveryData != null ? mTransportDiscoveryData.length : 0);
+        if (mTransportDiscoveryData != null) {
             dest.writeByteArray(mTransportDiscoveryData);
         }
     }
-- 
2.28.0

